From 540a23aaf66dc2df73dcfe76efd71e62bfe20d63 Mon Sep 17 00:00:00 2001
From: Jayachandran C <jayachandranc@google.com>
Date: Thu, 11 Jan 2018 14:10:17 +0200
Subject: [PATCH 14/20] legacy: allow rild to access radio data files

This rule has been removed from system public policy
but is still needed for old blobs.

Test: Verified radio works fine and no denials

Bug: 36740743
Change-Id: I2baefddf49194bb5b099e56d7b37623b99b6078f
---
 legacy-common/rild.te | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/legacy-common/rild.te b/legacy-common/rild.te
index ef86beb..3b0338d 100644
--- a/legacy-common/rild.te
+++ b/legacy-common/rild.te
@@ -1,2 +1,9 @@
 binder_use(rild)
 allow rild { mediaserver_service audioserver_service }:service_manager find;
+
+# rild reads qcril.db linked to /system/etc/qcril.db
+allow rild radio_data_file:lnk_file read;
+
+# allow rild to access radio data file
+allow rild radio_data_file:dir rw_dir_perms;
+allow rild radio_data_file:file create_file_perms;
-- 
2.7.4

